src/syscall_sdl.c

```diff
@@ -7,13 +7,76 @@
 #error "Do not manage to build this file unless you enable SDL support."
 #endif
 
+#include <pthread.h>
 #include <stdint.h>
 #include <stdio.h>
+#include <unistd.h>
 
 #include <SDL.h>
+#include <SDL_mixer.h>
 
 #include "state.h"
 
+extern uint8_t *mus2midi(uint8_t *, int *);
+
+/*
+ * In DOOM1.WAD only DSITMBK sound effect use sample rate 22050
+ * but normally only head in multiplayer
+ * since we are single player, stick with 11025 is OK
+ *
+ * In QUAKE, most sound effects with sample rate 11025
+ */
+#define SAMPLE_RATE 11025
+
+/* Most audio device supports stereo */
+#define CHANNEL_USED 2
+
+#define CHUNK_SIZE 2048
+
+#define MUSIC_MAX_SIZE 65536
+
+/* Max size of sound is around 18000 bytes */
+#define SFX_SAMPLE_SIZE 32768
+
+/* sound-related request type */
+enum {
+    INIT_AUDIO,
+    SHUTDOWN_AUDIO,
+    PLAY_MUSIC,
+    PLAY_SFX,
+    SET_MUSIC_VOLUME,
+    STOP_MUSIC,
+};
+
+typedef struct sound {
+    uint8_t *data;
+    size_t size;
+    int looping;
+    int volume;
+} sound_t;
+
+/* SDL-mixer-related and music-related variables */
+static pthread_t music_thr;
+static uint8_t *mid_data;
```

Comment:
Rename it to `midi_data`. The original term `mid_data` might lead to confusion, as it could be interpreted as either "middle data" or "data in the middle."